net/zerotier/src/opnsense/mvc/app/controllers/OPNsense/Zerotier/Api/ZerotierController.php

```diff
@@ -45,12 +45,26 @@ public function getAction()
     {
         $result = array();
         if ($this->request->isGet()) {
-            $mdlZerotier = new Zerotier();
+            $mdlZerotier = $this->getModel();
             $result['zerotier'] = $mdlZerotier->getNodes();
         }
         return $result;
     }
 
+    public function setAction()
+    {
+        $result = array("result" => "failed");
+        if($this->request->isPost()) {
+            $mdlZerotier = $this->getModel();
+            $mdlZerotier->setNodes($this->request->getPost("zerotier"));
+            $mdlZerotier->serializeToConfig();
+            Config::getInstance()->save();
+            $enabled = $this->isEnabled($mdlZerotier);
+            $result["result"] = $this->toggleZerotierService($enabled);
```

Comment:
This looks strange. Why is it toggled after saving?

Reply:
For safety. I persist the model to config.xml first, then I toggle the network on (or off). Would you recommend a different approach?

Reply:
no, I found out that the method name is misleading

Reply:
in the programming sense, maybe, not natural language... explicitly toggle on/off is ok, the $enabled makes that clear.